test/parallel/test-net-write-after-close.js

```diff
@@ -21,24 +21,27 @@
 
 'use strict';
 const common = require('../common');
+
 const net = require('net');
 
+let serverSocket;
+
 const server = net.createServer(common.mustCall(function(socket) {
+  serverSocket = socket;
+
   socket.resume();
 
   socket.on('error', common.mustCall(function(error) {
     console.error('got error, closing server', error);
     server.close();
   }));
-
-  setTimeout(common.mustCall(function() {
-    console.error('about to try to write');
-    socket.write('test', common.mustCall());
-  }), 250);
 }));
 
 server.listen(0, function() {
   const client = net.connect(this.address().port, function() {
+    client.on('end', () => {
```

Comment:
Actually now that I read this better, `'end'` is correct.

`'finish'` is part of the streams API, but it will signal when the current side is closed. `'end'`  will be emitted when the other side as stopped sending, so when we call `serverSocker.write()` that will lead to an error.